test/distributed/fsdp/test_fsdp_state_dict.py

```diff
@@ -1236,7 +1236,7 @@ def test_world_size_one(self):
 class TestFSDPStateDict4GPUs(FSDPTest):
     @property
     def world_size(self):
-        return max(torch.cuda.device_count(), 2)
+        return torch.cuda.device_count()
```

Comment:
It looks like these tests have hardcoded batch size 4 and have logic like `if rank < 2`, would it work if we don't constrain the max to 2 and min to `torch.cuda.device_count`?

Reply:
I think this is okay. The batch size is the local batch size not the global one. Should not be affected by the rank size. This line should work

Reply:
Yes this would a local change. In the case where only CPU is available, it would skip the test with `[2024-02-15 23:36:14,514] torch.testing._internal.common_distributed: [WARNING] Note: no subprocesses were spawned, test was likely skipped.`. And in the case where there are more than 2 GPUs, we should not run into error.